core/include/detray/intersection/line_intersector.hpp

```diff
@@ -85,8 +64,8 @@ struct ray_line_intersector {
         const scalar zd = vector::dot(_z, _d);
 
         // Case for wire is parallel to track
-        if (1 - std::abs(zd) < 1e-5) {
-            return intersection_type{};
+        if (1 - static_cast<detray::scalar>(std::abs(zd)) < 1e-5) {
```

Comment:
If ```zd``` is a ```detray::scalar```, the return value of ```std::abs``` is also ```detray::scalar```, if I am not mistaken. Only in the operation ```1 - std::abs(zd)``` it will be cast to ```double```. In my previous comment that was what I was confused about, but I think that I was stupid, because that one cast from ```float``` to ```double``` if ```detray::scalar``` is ```float``` should not change much in the end, right?

Reply:
I will use zd*zd

Reply:
Writing suggestion would be helpful cause I am ignorant of cast stuff...

Reply:
Oh, really, sorry! Yeah, my writing is not clear sometimes. This is my understanding, which is also limited: ```std::abs(zd)``` will be of type ```detray::scalar```, so either ```float``` or ```double```. The ```1``` in ```1 - std::abs(zd)``` is an integer, so will be converted to ```detray::scalar``` to do the subtraction. The result will then be compared to ```1e-5``` which is a double, so the result of the subtraction will be converted to ```double``` before the comparison.

I would write this as ```scalar{1.} - std::abs(zd) < scalar{1e-5}```, but that is just to the level of my understanding...